polyfill/lib/intl.mjs

```diff
@@ -27,37 +31,74 @@ const CAL_ID = Symbol('calendar-id');
 const LOCALE = Symbol('locale');
 const OPTIONS = Symbol('options');
 
-const descriptor = (value) => {
-  return {
-    value,
-    enumerable: true,
-    writable: false,
-    configurable: true
-  };
-};
-
 const IntlDateTimeFormat = globalThis.Intl.DateTimeFormat;
 const ObjectAssign = Object.assign;
+const ObjectCreate = Object.create;
+const ObjectDefineProperty = Object.defineProperty;
 
 // Construction of built-in Intl.DateTimeFormat objects is sloooooow,
 // so we'll only create those instances when we need them.
 // See https://bugs.chromium.org/p/v8/issues/detail?id=6528
-function getPropLazy(obj, prop) {
-  let val = obj[prop];
+function getSlotLazy(obj, slot) {
+  let val = GetSlot(obj, slot);
   if (typeof val === 'function') {
-    val = new IntlDateTimeFormat(obj[LOCALE], val(obj[OPTIONS]));
-    obj[prop] = val;
+    val = new IntlDateTimeFormat(GetSlot(obj, LOCALE), val(GetSlot(obj, OPTIONS)));
+    SetSlot(obj, slot, val);
   }
   return val;
 }
 
-export function DateTimeFormat(locale = undefined, options = undefined) {
-  if (!(this instanceof DateTimeFormat)) return new DateTimeFormat(locale, options);
+function createDateTimeFormat(dtf, locale, options) {
   const hasOptions = typeof options !== 'undefined';
-  options = hasOptions ? ObjectAssign({}, options) : {};
+  if (hasOptions) {
+    // Read all the options in the expected order and copy them to a
+    // null-prototype object with which we can do further operations
+    // unobservably
+    const props = [
+      'localeMatcher',
+      'calendar',
+      'numberingSystem',
+      'hour12',
+      'hourCycle',
+      'timeZone',
+      'weekday',
+      'era',
+      'year',
+      'month',
+      'day',
+      'dayPeriod',
+      'hour',
+      'minute',
+      'second',
+      'fractionalSecondDigits',
+      'timeZoneName',
+      'formatMatcher',
+      'dateStyle',
+      'timeStyle'
+    ];
+    options = ES.ToObject(options);
+    const newOptions = ObjectCreate(null);
+    for (const prop of props) {
+      if (ES.HasOwnProperty(options, prop)) {
+        // Avoid triggering setters on Object.prototype
```

Comment:
How would it trigger them with a null prototype?

Reply:
Right. That must have been left over from some intermediate debugging.